### PR TITLE
[erts|esock] Socket accept leaks monitors

### PR DESCRIPTION
**Pull request by bmk**
Add missing demonitor after a successful accept.
Also added monitor "init" after a demonitor (to make sure nothing remains of the previous monitor).

OTP-18240

### CODE REVIEW

File: erts/emulator/nifs/common/prim_socket_nif.c [hunk 1]
```diff
@@ -7001,6 +7021,12 @@ ERL_NIF_TERM esock_accept_accepting_current_accept(ErlNifEnv*       env,
     if (esock_accept_accepted(env, descP, sockRef, accSock,
                               descP->currentAcceptor.pid, &res)) {
 
+        ESOCK_ASSERT( DEMONP("esock_accept_accepting_current_accept -> "
+                             "current acceptor",
+                             env, descP, &descP->currentAcceptor.mon) == 0);
+
+        MON_INIT(&descP->currentAcceptor.mon);
```
Comment (max-au) [inline erts/emulator/nifs/common/prim_socket_nif.c:7028]: Is it necessary to `MON_INIT` after demonitoring is done?

**Review by max-au — COMMENTED** [review]
Looks okay to me, do you want to add test code from the bug report?

File: erts/emulator/nifs/common/prim_socket_nif.c (hunk 1 above)
Comment (bmk) [reply erts/emulator/nifs/common/prim_socket_nif.c:7028]: I have taken the test code and added it to socket_SUITE (with some minor modifications).
MON_INIT is mostly *really* sure...

**Review by max-au — APPROVED** [review]
This fixes the problem.
For test code, I guess @RaimoNiskanen may have some input.

**Comment by bmk** [comment]
Merged into maint (and master) and will be included in a patch to be release later this week (for maint-25).

**Pull request closed without merging**